### PR TITLE
Improve Model Accuracy How-To

### PR DESCRIPTION
**Pull request by luisquintanilla**
Improve Model Accuracy How-To

[Internal Review Link](https://review.docs.microsoft.com/en-us/dotnet/machine-learning/resources/improve-machine-learning-model-ml-net?branch=pr-en-us-12031)

[AB#1450827](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1450827)

### CODE REVIEW

File: docs/machine-learning/how-to-guides/improve-machine-learning-model-ml-net.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: "How-To: Improve model accuracy"
+description: Learn how to improve model accuracy
+ms.date: 04/25/2019
+author: luisquintanilla
+ms.author: luquinta
+ms.custom: mvc,how-to
```
Comment (JRAlexander) [inline docs/machine-learning/how-to-guides/improve-machine-learning-model-ml-net.md]: add "ms.topic: how-to" and remove how-to from ms.custom

**Review by JRAlexander — CHANGES_REQUESTED** [review]
Just a metadata change then you are good to go.

File: docs/machine-learning/how-to-guides/improve-machine-learning-model-ml-net.md (hunk 1 above)
Comment (JRAlexander) [inline docs/machine-learning/how-to-guides/improve-machine-learning-model-ml-net.md]:
```suggestion
ms.custom: mvc
ms.topic: how-to
```

**Review by JRAlexander — APPROVED** [review]

**Comment by CESARDELATORRE** [comment]
All those points are good ones when trying to improve accuracy. But we’re missing the most important, easiest and more productive way:

- “Use the ML.NET CLI / AutoML”
(Currently supported only for binary and multi class classification and regression, though)

That’s the first thing I’d try.

However, this is complementary with other approaches you mentioned like larger datasets, etc. You might want to try multiple approaches at the same time.

File: docs/machine-learning/resources/improve-machine-learning-model-ml-net.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## Provide more data samples
+
+Like humans, the more training algorithms get, the likelihood of better performance increases. One way to improve model performance is to provide more training data samples to the algorithms. The more data it learns from, the more cases it is able to correctly identify. 
+
+## Add context to the data
+
+The meaning of a single data point can be difficult to interpret. Building context around the data points helps algorithms as well as subject matter experts better make decisions. For example, the fact that a house has three bedrooms does not on its own give a good indication of its price. However, if you add context and now know that it is in a suburban neighborhood outside of a major metropolitan area where media age is 38, median household income is $80,000 and schools are in the top 20th percentile then the algorithm has more information to base its decisions on. All of this context can be added as input to the machine learning as features.
```
Comment (sfilipi) [inline docs/machine-learning/resources/improve-machine-learning-model-ml-net.md]: >media [](start = 413, length = 5)

average, median

File: docs/machine-learning/resources/improve-machine-learning-model-ml-net.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Use meaningful data and features
+
+Although more data samples and features can help improve the accuracy of the model, they may also introduce noise since not all data and features are meaningful. Therefore, it is important to understand which features are the ones that most heavily impact decisions made by the algorithm. Using techniques like Permutation Feature Importance (PFI) can help identify those salient features and not only help explain the model but also use the output as a feature selection method to reduce the amount of noisy features going into the training process. 
+
+Learn to use PFI by visiting the following [link](./explain-machine-learning-model-permutation-feature-importance-ml-net.md)
+
+## Cross-validation
+
+Cross-validation is a training and model evaluation technique that splits the data into several partitions and trains multiple algorithms on these partitions. This technique improves the robustness of the model by holding out data from the training process. In addition to improving performance on unseen observations, in data constrained environments it can be an effective tool for training models with a smaller dataset.
```
Comment (sfilipi) [inline docs/machine-learning/resources/improve-machine-learning-model-ml-net.md]: >data constrained [](start = 322, length = 16)

for my own education, does this need a dash: data-constrained?

**Review by sfilipi — APPROVED** [review]
:shipit:

File: docs/machine-learning/resources/improve-machine-learning-model-ml-net.md (hunk 2 above)
Comment (luisquintanilla) [reply docs/machine-learning/resources/improve-machine-learning-model-ml-net.md]: #resolved

File: docs/machine-learning/resources/improve-machine-learning-model-ml-net.md (hunk 3 above)
Comment (luisquintanilla) [reply docs/machine-learning/resources/improve-machine-learning-model-ml-net.md]: #resolved

**Review by natke — APPROVED** [review]

**Pull request merged**